ONSdigital/dp-elasticsearch
# improve error handling in v710 client

### What

- Fixed nil pointers in some cases where `res.StatusCode` was being called with nil res 
- Extend error message when `res.IsError()`, by reading the response body - this helps identifying the root cause of the error.
- Fixed error checking in healthchecks
- Fixed deferred func inside loop (fixed by creating an inner function with the defer in it instead)

Maybe a further improvement would be to use `dp-net/error` package, but maybe this approach should be discussed with the team and/or TLs first.

Changes tested while running re-indexing in search api as part of [this spike](https://trello.com/c/30uwopFT/1233-spike-investigate-what-the-data-model-should-be-for-storing-dimensions-and-population-types-in-es) , note that there isn't any other PR for the spike yet

### How to review

- Make sure code changes make sense
- Make sure tests are successful

### Who can review

Anyone
